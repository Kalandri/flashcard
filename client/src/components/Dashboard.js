import React from 'react';
import { Link } from 'react-router-dom';
import StackList from './StackList';
import Header from './Header';

const Dashboard = () => {
  return (
    <div>
      <StackList />
      <hr />
      <Link to='/stackform'><h4>Create a New Card Stack</h4></Link>
    </div>
  )
}

export default Dashboard;