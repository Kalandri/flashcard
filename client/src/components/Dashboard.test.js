import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import rootReducer from '../reducers';

import { shallow } from 'enzyme';
import Dashboard from './Dashboard';
import '../setupTests.js';

describe ('Dashboard', ()=>{
	const dashboard = shallow(<Dashboard/>);
	it('renders the Stacklist', ()=>{
		expect(dashboard.find('Connect(StackList)').exists()).toBe(true);
	});

	it('render a link to create a new stack', ()=>{
		expect(dashboard.find('Link h4').text()).toEqual('Create a New Card Stack');
	});
});

test('snapshot test for Dashboard', ()=>{
	const store = createStore(rootReducer);
	const component =renderer.create(
		<Provider store={store}>
		<MemoryRouter>
			<Dashboard />
		</MemoryRouter>
		</Provider>
	);
	const tree=component.toJSON();
	expect(tree).toMatchSnapshot();
})


