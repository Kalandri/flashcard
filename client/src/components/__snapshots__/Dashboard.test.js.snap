// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test for Dashboard 1`] = `
<div>
  <div>
    <h3>
      List of Flash Cards
    </h3>
    <a
      href="/stack"
      onClick={[Function]}
    >
      <h4>
        State Abbreviations
      </h4>
    </a>
    <a
      href="/stack"
      onClick={[Function]}
    >
      <h4>
        Words For Computer Science Basics
      </h4>
    </a>
  </div>
  <hr />
  <a
    href="/stackform"
    onClick={[Function]}
  >
    <h4>
      Create a New Card Stack
    </h4>
  </a>
</div>
`;
